### PR TITLE
feat: use slack for career form

### PR DESCRIPTION
**Pull request by hofmann-felix**
### What's included?
* Applications are now sent to a private Slack channel instead of Personio.
* For the implementation (similar to the contact form) Gatsby Functions and the Slack web api are used
* **Note for reviewers**: If you need access to the Slack channel please let me now

### Preview
* URL: https://satellytescommain21751-featslackapplications.gatsbyjs.io/de/career/frontend-engineer/
* Screenshot of an application inside the slack channel:
Bildschirm­foto 2023-03-15 um 14 37 26

### CODE REVIEW

File: src/components/pages/career-details/career-details.tsx [hunk 1]
```diff
@@ -42,13 +35,6 @@ export const CareerDetails = ({
     ref?.current?.scrollIntoView({ behavior: 'smooth', block: 'start' });
   };
 
-  const location = useLocation();
-  const searchParams = new URLSearchParams(location.search);
-  const personioChannel = searchParams.get('_pc');
-  const utmChannel = UTM_CHANNEL_MAP[searchParams.get('utm_source') ?? ''];
-
-  const recruitingChannel = utmChannel ?? personioChannel ?? DEFAULT_CHANNEL_ID;
-
```
Comment (hofmann-felix) [inline src/components/pages/career-details/career-details.tsx:56]: Do we still need information about the recruitingChannel or is this related to personio?

File: src/api/career-form.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        message,
+        location,
+        available_from,
+        salary_expectations,
+        jobName,
+        files,
+      }),
+      icon_emoji: ':rocket:',
+    });
+    console.log(postMessageResult);
```
Comment (georgiee) [inline src/api/career-form.ts]: remove and check other occurences 👍

File: src/api/career-form.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    for (const file of files) {
+      const fileUploadResult = await client.files.upload({
```
Comment (georgiee) [inline src/api/career-form.ts]: pass message id of the actual message that you posted before as threadTs to create a thread. Will be more tidy.
https://api.slack.com/methods/files.upload

File: src/components/pages/career-details/career-details.tsx (hunk 1 above)
Comment (georgiee) [reply src/components/pages/career-details/career-details.tsx:56]: Yes that would be could. We never implemented that on the website though. So remove it here and then let's come back. I would like to add a parameter to our job postings so we can share it on different channels and then know through which channel someone came. Could be website, email etc or also platform or person related. Let's discuss this on a separate PR though

File: src/components/pages/career-details/new-career-form/submit-application.ts [hunk 4]
```diff
@@ -72,40 +58,25 @@ export const submitApplication = async (
...
-      const personioErrorMessage =
-        error?.response?.data?.error?.message ??
-        error?.response?.data?.error ??
-        error.message;
-
+      console.error(error);
       setError('api', {
         type: 'server',
-        message: `Irgendetwas ist schief gelaufen (${personioErrorMessage}).`,
+        message: 'Slack API error',
```
Comment (georgiee) [inline src/components/pages/career-details/new-career-form/submit-application.ts]: I would not expose that we send it to slack. Just say API Error.

**Review by georgiee — CHANGES_REQUESTED** [review]
Thanks it works!

I found two issues in the form:
1. "Datenschutz" Link at the bottom opens Personio
2. I can't click to trigger the file picker. I have to drag and drop (Firefox). Can you confirm that?

File: src/api/career-form.ts (hunk 3 above)
Comment (hofmann-felix) [reply src/api/career-form.ts]: Good idea 👍 Looks much better now inside the Slack channel

**Comment by hofmann-felix** [comment]
@georgiee Thanks for your review, I implemented your suggestions 👍

I also encountered the file dropper problem in Firefox (also on satellytes.com). Here is the fix for it: https://github.com/satellytes/satellytes.com/pull/736

File: data/pages/data-privacy-de.md [hunk 5]
```diff
@@ -9,8 +9,8 @@ language: de
    Daten sind alle Daten, die auf Sie persönlich beziehbar sind, z. B. Name, Adresse, E-Mail-Adressen, Nutzerverhalten.
 2. Verantwortlicher gem. Art. 4 Abs. 7 EU-Datenschutz-Grundverordnung (DSGVO) ist Satellytes Digital Consulting GmbH,
    Sendlinger Straße 52, 80331 München, info@satellytes.com (siehe [Impressum](/imprint/)).
-3. Bei Ihrer Kontaktaufnahme mit uns per E-Mail oder über ein Kontaktformular werden die von Ihnen mitgeteilten Daten (
-   Ihre E-Mail-Adresse, ggf. Ihr Name und Ihre Telefonnummer) von uns gespeichert, um Ihre Fragen zu beantworten. Die in
+3. Bei Ihrer Kontaktaufnahme mit uns per E-Mail oder über ein Kontaktformular werden die von Ihnen mitgeteilten Daten 
+   (Ihre E-Mail-Adresse, Ihr Vorname, ggf. Ihr Nachname und ggf. Ihr Lebenslauf) von uns gespeichert, um Ihre Fragen zu beantworten. Die in
```
Comment (hofmann-felix) [inline data/pages/data-privacy-de.md]: @georgiee It would be great if you could check if this is ok (also the English Version)

File: data/pages/data-privacy.md [hunk 6]
```diff
@@ -9,8 +9,8 @@ language: en
    data are all data that can be related to you personally, e.g. name, address, e-mail addresses, user behavior.
 2. The responsible party pursuant to Article 4 (7) of the EU General Data Protection Regulation (GDPR) is Satellytes Digital Consulting GmbH,
    Sendlinger Straße 52, 80331 Munich, Germany, info@satellytes.com (see [Imprint](/imprint/)).
-3. When you contact us by e-mail or via a contact form, the data you provide (
-   your e-mail address, if applicable your name and your telephone number) will be stored by us in order to answer your questions. The data
+3. When you contact us by e-mail or via a contact form, the data you provide 
+   (your e-mail address, your first name, if applicable your last name and your CV) will be stored by us in order to answer your questions. The data
```
Comment (georgiee) [inline data/pages/data-privacy.md]:
```suggestion
   (your e-mail address, your first name, if applicable your last name and your CV or other uploaded files) will be stored by us in order to answer your questions. The data
```

File: data/pages/data-privacy-de.md (hunk 5 above)
Comment (georgiee) [inline data/pages/data-privacy-de.md]:
```suggestion
   (Ihre E-Mail-Adresse, Ihr Vorname, ggf. Ihr Nachname oder Ihr Lebenslauf bzw. andere hochgeladene Dateien) von uns gespeichert, um Ihre Fragen zu beantworten. Die in
```

**Review by georgiee — APPROVED** [review]
Thanks for fixing the two issues around privacy policy and the firefox bug (in the other PR). 
Let's merge this 🙌
